flink-runtime/src/test/java/org/apache/flink/runtime/util/SerializedThrowableTest.java

```diff
@@ -179,4 +188,62 @@ void testCopySuppressed() {
                 .isInstanceOf(SerializedThrowable.class)
                 .hasMessage("java.lang.Exception: suppressed");
     }
+
+    @Test
+    void testCyclicSuppressedThrowableSerialized() {
+        SerializedThrowable serializedThrowable = new SerializedThrowable(mockThrowable());
+        assertThat(serializedThrowable).isNotNull();
+    }
+
+    // Use the repeated test for checking the thread safe and stable.
+    @RepeatedTest(16)
+    void testCyclicSuppressedThrowableConcurrentSerialized() throws InterruptedException {
```

Comment:
I think this test should add a timeout limit (maybe 5 seconds) as the original test is used to detect dead lock, which would cause the test never quit.